File: Backend/ev-booking-sys/src/main/java/com/hanseul/ev_booking_sys/model/dto/ApiResponse.java
package com.hanseul.ev_booking_sys.model.dto;

import com.hanseul.ev_booking_sys.exception.ResponseCode;

import java.time.LocalDateTime;

public record ApiResponse<T>(
        int code,
        String message,
        T data,
        LocalDateTime timeStamp
) {
    public static <T> ApiResponse<T> success(ResponseCode code, T data) {
        return new ApiResponse<>(code.getCode(), code.getMessage(), data, LocalDateTime.now());
    }

    public static <T> ApiResponse<T> error(ResponseCode code, String detail) {
        return new ApiResponse<>(code.getCode(), detail, null, LocalDateTime.now());
    }
}

File: Android/core/model/src/main/java/com/hs/model/ApiResponse.kt
package com.hs.model

import com.google.gson.annotations.SerializedName

data class ApiResponse<T>(
    @SerializedName("code")
    val code: Int,
    @SerializedName("message")
    val message: String,
    @SerializedName("data")
    val data: T?,
    @SerializedName("timeStamp")
    val timeStamp: String
) {
    val isSuccess: Boolean
        get() = code in 200..299
}
